Classes/Eel/ElasticSearchQueryBuilder.php

```diff
@@ -758,6 +758,8 @@ public function moreLikeThis(array $like, array $fields = [], array $options = [
      */
     public function query(NodeInterface $contextNode)
     {
+        $this->elasticSearchClient->setDimensions($contextNode->getContext()->getTargetDimensions());
```

Comment:
This parts is the weekiest code of the PR ... you MUST do that before querying to switch the the correct index. I will try to design a `ÌndexSelectorStrategyInterface` in the `Flowpack.ElasticSearch` package, to have a central place where you can configure how the switch is done. But I think a bit of sleep before ;)

Reply:
I think I have a solution `ElasticSearchClient` MUST store the context node (actually the context node is stored in the QueryBuilder, like this when calling `$this->elasticSearchClient->getIndex()->request()` we can switch the index. In the case the context node is required, and we need to test feature, like count, more like this....

Reply:
Done in #80a2e152c117ac90c91deeb82eb177fa3650ee23